### ISSUE TITLE
ARRAYCONTAINS with JSON Array generates error

### ISSUE THREAD

**Issue by tpanagos** [confluentinc/ksql#1960]
KSQL-CLI 5.0.0 using a message with a JSON body. I cannot find a format for using ARRAYCONTAINS that returns as a result. I suspect that this issue is related to #1562. I notice that the array values after creating a stream no longer have surrounding "". 

`CREATE STREAM report_four (obj STRUCT<tags ARRAY<STRING>>) WITH (kafka_topic='create-json', value_format='JSON');`

```
ksql> DESCRIBE REPORT_FOUR;

Name                 : REPORT_FOUR
 Field   | Type                                
-----------------------------------------------
 ROWTIME | BIGINT           (system)           
 ROWKEY  | VARCHAR(STRING)  (system)           
 OBJ     | STRUCT<TAGS ARRAY<VARCHAR(STRING)>> 
-----------------------------------------------
For runtime statistics and query details run: DESCRIBE EXTENDED <Stream,Table>;
```

```
 SELECT OBJ->TAGS FROM report_four;
[microshare, location, battery]
[PTW, Test, 355, Tour Bus, public]
```
```
ksql> SELECT * FROM report_four WHERE ARRAYCONTAINS(OBJ->TAGS,'microshare');
Failed to generate code for filterExpression:ARRAYCONTAINS(FETCH_FIELD_FROM_STRUCT(REPORT_FOUR.OBJ, 'TAGS'), 'microshare') schema:org.apache.kafka.connect.data.SchemaBuilder@56c8e169
Caused by: Failed to create instance of kudfClass class io.confluent.ksql.function.udf.json.ArrayContainsKudf for function ARRAYCONTAINS
Caused by: Class io.confluent.ksql.function.KsqlFunction can not access a member of class io.confluent.ksql.function.udf.json.ArrayContainsKudf with modifiers ""
ksql> SELECT * FROM report_four WHERE ARRAYCONTAINS(OBJ->TAGS,"microshare");
Column microshare cannot be resolved.
ksql> SELECT * FROM report_four WHERE ARRAYCONTAINS(OBJ->TAGS,MICROSHARE);
Column MICROSHARE cannot be resolved.
ksql> SELECT ARRAYCONTAINS(OBJ->TAGS,'microshare') FROM report_four;
Unexpected error generating code for SelectValueMapper
Caused by: Failed to create instance of kudfClass class io.confluent.ksql.function.udf.json.ArrayContainsKudf for function ARRAYCONTAINS
Caused by: Class io.confluent.ksql.function.KsqlFunction can not access a member of class io.confluent.ksql.function.udf.json.ArrayContainsKudf with modifiers ""
```

Perhaps I am misusing ARRAYCONTAINS. There isn't much in the way of documentation or test cases for this scalar.

**Comment by apurvam** [comment]
It may well be related to #1562 . Now that it is fixed, you can try doing an `ARRAYCONTAINS` on the array extracted using `EXTRACTJSONFIELD`. I haven't dug into the implementation of array contains, but the [docs](https://docs.confluent.io/current/ksql/docs/developer-guide/syntax-reference.html#scalar-functions) indicate that it only works with JSON arrays.

**Comment by rmoff** [comment]
Reproduced in 5.1. 

Load some test data: 

    curl -s "https://api.mockaroo.com/api/e4e65ec0?count=15&key=ff7856d0" | \
    kafkacat -b localhost:9092 -t nested_array_test -P

Create stream 

```
ksql> CREATE STREAM NESTED_ARRAY_TEST (id BIGINT, first_name VARCHAR, last_name VARCHAR, email VARCHAR, preferred STRUCT<colours ARRAY<VARCHAR>>) WITH (kafka_topic='nested_array_test', value_format='JSON');

 Message
----------------
 Stream created
----------------
```

Query the stream

```
ksql> SELECT FIRST_NAME, PREFERRED->COLOURS FROM NESTED_ARRAY_TEST;
Bethena | [Violet, Maroon, Yellow]
Bartholomew | [Aquamarine, Crimson]
Felizio | [Red]
Emmott | [Teal]
Elianore | [Violet, Pink]
```

Attempt the `ARRAYCONTAINS` function

```
ksql> SELECT FIRST_NAME, PREFERRED->COLOURS FROM NESTED_ARRAY_TEST WHERE ARRAYCONTAINS(PREFERRED->COLOURS,'Pink');
Failed to generate code for filterExpression:ARRAYCONTAINS(FETCH_FIELD_FROM_STRUCT(NESTED_ARRAY_TEST.PREFERRED, 'COLOURS'), 'Pink') schema:org.apache.kafka.connect.data.SchemaBuilder@76ea44ac
Caused by: Failed to create instance of kudfClass class io.confluent.ksql.function.udf.json.ArrayContainsKudf for function ARRAYCONTAINS
Caused by: Class io.confluent.ksql.function.KsqlFunction can not access a member of class io.confluent.ksql.function.udf.json.ArrayContainsKudf with modifiers ""
```
